crt_portal/cts_forms/filters.py

```diff
@@ -61,12 +61,16 @@ def report_filter(request):
             elif filter_options[field] == '__contains':
                 kwargs[f'{field}__icontains'] = request.GET.getlist(field)[0]
             elif 'date' in field:
-                # filters by a start date or an end date expects YYYYMMDD
+                # filters by a start date or an end date expects yyyy-mm-dd
                 field_name = _get_date_field_from_param(field)
-                year = int(request.GET.getlist(field)[0][:4])
-                month = int(request.GET.getlist(field)[0][4:6])
-                day = int(request.GET.getlist(field)[0][6:])
-                kwargs[f'{field_name}{filter_options[field]}'] = datetime.date(year, month, day)
+                encodedDate = request.GET.getlist(field)[0]
+                decodedDate = urllib.parse.unquote(encodedDate)
+                dateObj = datetime.strptime(decodedDate, "%Y-%m-%d")
```

Comment:
Can we check for invalid input here? If the user enters invalid input, we'll get something like: `time data '01-20-2020' does not match format '%Y-%m-%d'` or `time data 'test' does not match format '%Y-%m-%d'`. This will result in a 500 page in production.

Reply:
Using try catch on strptime, settiing ill-formatted dates to default of 01-01-2000.
```code
try:
                    dateObj = datetime.strptime(decodedDate, "%Y-%m-%d")
                except ValueError:
                    # if the date is invalid, set it to 2020 because there is no data prior to 2020
                    dateObj = datetime.strptime("2020", "%Y")
```